backend/src/apiserver/server/util.go

```diff
@@ -54,20 +54,20 @@ func loadFile(fileReader io.Reader, maxFileLength int) ([]byte, error) {
 	return pipelineFile[:size], nil
 }
 
-func isSupportedPipelineFormat(fileName string) bool {
-	return isYamlFile(fileName) || isCompressedTarballFile(fileName) || isZipFile(fileName)
+func isSupportedPipelineFormat(fileName string, compressedFile []byte) bool {
+	return isYamlFile(fileName) || isCompressedTarballFile(compressedFile) || isZipFile(compressedFile)
 }
 
 func isYamlFile(fileName string) bool {
 	return strings.HasSuffix(fileName, ".yaml") || strings.HasSuffix(fileName, ".yml")
 }
 
-func isZipFile(fileName string) bool {
-	return strings.HasSuffix(fileName, ".zip")
+func isZipFile(compressedFile []byte) bool {
+	return len(compressedFile) > 2 && compressedFile[0] == '\x50' && compressedFile[1] == '\x4B' //Signature of zip file is "PK"
```

Comment:
could you add a test to see what would happen when a plain file starts with PK?

Reply:
I can, but I think it's already covered by the `TestDecompressPipelineZip_MalformattedZip` test. It's just a broken archive.

Reply:
the file is empty. could you create a file.txt with "PK" in the file?

Reply:
I've added another test. https://github.com/kubeflow/pipelines/pull/919/commits/a3c4715be3151bbbe7d61454101830709d2c874a